third_party/packages/ansi_up/lib/ansi_up.dart

```diff
@@ -247,9 +250,9 @@ class AnsiUp {
             }
           }
           if (modeCmd == '2' && index + 2 < sgrCmds.length) {
-            final r = int.tryParse(sgrCmds[index++], radix: 10);
-            final g = int.tryParse(sgrCmds[index++], radix: 10);
-            final b = int.tryParse(sgrCmds[index++], radix: 10);
+            final r = int.tryParse(sgrCmds[index++], radix: 10)!;
+            final g = int.tryParse(sgrCmds[index++], radix: 10)!;
+            final b = int.tryParse(sgrCmds[index++], radix: 10)!;
```

Comment:
the original Dart code had a bug when porting from JS. The intent is to skip treating as a color if the input is invalid not to blow up.
Remove the `!` and add
`r != null && g != null && b != null`
to the checks.
bonus points, add a test that verifies it does not blowup on invalid input.